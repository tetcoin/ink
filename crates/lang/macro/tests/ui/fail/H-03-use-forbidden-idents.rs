use pro_lang as pro;

#[pro::contract]
mod forbidden_indents {
    #[pro(storage)]
    pub struct ForbiddenIndents {}

    impl ForbiddenIndents {
        #[pro(constructor)]
        pub fn constructor() -> Self {
            Self {}
        }

        /// An pro! message starting with __pro_ prefix.
        #[pro(message)]
        pub fn __pro_message(&self) {
            // All identifiers starting with `__pro_` are forbidden to use in pro!.
            let __pro_first = ();
        }
    }
}

fn main() {}
