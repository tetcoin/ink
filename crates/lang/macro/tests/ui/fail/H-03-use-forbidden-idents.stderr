error: encountered invalid identifier starting with __pro_
  --> $DIR/H-03-use-forbidden-idents.rs:16:16
   |
16 |         pub fn __pro_message(&self) {
   |                ^^^^^^^^^^^^^

error: encountered invalid identifier starting with __pro_
  --> $DIR/H-03-use-forbidden-idents.rs:18:17
   |
18 |             let __pro_first = ();
   |                 ^^^^^^^^^^^
